core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -343,6 +359,14 @@ public ReadableByteChannel get(PageId pageId, int pageOffset) {
         return null;
       }
       ReadableByteChannel ret = getPage(pageId, pageOffset);
+      if (ret == null) {
+        // something is wrong to read this page, let's remove it from meta store
+        try (LockResource r2 = new LockResource(mMetaLock.writeLock())) {
+          mMetaStore.removePage(pageId);
+        } catch (PageNotFoundException e) {
+          // best effort to remove this page from meta store and ignore the exception
+        }
+      }
```

Comment:
Consider putting this logic into a method?

Reply:
Yeah I would have thought of this an integral logic of getPage method.

Reply:
I considered that too initially. but one reason I still leave this in `get()` is to make all locking related code in one method --- in here we are acquiring `mMetaLock`. It can be harder to track if somewhere else also acquire locks